### ISSUE TITLE
provide stack installation method

### ISSUE THREAD

**Issue by m0davis** [cedille/cedille#52]
I use stack to install agda rather than cabal because of "cabal hell" (as it is known in the literature). The same fate grips me when trying to install cedille. Would you provide a stack.yaml for cedille, such as is provided for in agda?

**Comment by astump** [comment]
Great idea!  Would you be so kind as to file an issue for this on our
github page for Cedille?  Thanks, Aaron

**Comment by cwjnkins** [comment]
Hi, @m0davis,

I don't have much experience creating cabal packages or stack projects, so the commit I just pushed is a bit of a hack. However, it seems like it should be enough to get you to compile Cedille until a more robust solution is found

After making a clean clone of Cedille and checking out branch `stack-hack', the magic incantation is

    stack build # pulls down dependencies, and Agda directly from Github
    (export PATH="./.stack-work/install/x86_64-linux/lts-9.21/8.0.2/bin/":$PATH ; make) # run make in a subshell to let it know where Agda can be found

Let me know if this works for you.

**Comment by m0davis** [comment]
@cwnkins, thanks. I managed to get it to work with a couple of adjustments. First, the path you gave was not quite right on my system (I needed `x86_64-linux-tinfo6`). Second, during agda's compilation phase, it called `ghc` rather than `stack ghc --`, so I ran into the same problems as referenced in the OP. To solve that, I created a bash script named "ghc" with contents `stack ghc -- $@` and placed it in a location early in my PATH. After that, I fought with emacs a while to get the cedille-mode to work, but now it does, so great!
